examples/esp_idf/fw_update/main/app_main.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "nvs.h"
+#include "shell.h"
+#include "wifi.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/semphr.h"
+#include <golioth/client.h>
+#include <golioth/fw_update.h>
+#include <string.h>
+
+#define TAG "fw_update"
+
+static const char *_current_version = "1.2.3";
+
+static SemaphoreHandle_t _connected_sem = NULL;
+
+static void on_client_event(struct golioth_client *client,
+                            enum golioth_client_event event,
+                            void *arg)
+{
+    bool is_connected = (event == GOLIOTH_CLIENT_EVENT_CONNECTED);
+    if (is_connected)
+    {
+        xSemaphoreGive(_connected_sem);
+    }
+    GLTH_LOGI(TAG, "Golioth client %s", is_connected ? "connected" : "disconnected");
+}
+
+void app_main(void)
+{
+    GLTH_LOGI(TAG, "Start FW Update sample.");
+
+    // Initialize NVS first.
+    nvs_init();
+
+    // Create a background shell/CLI task (type "help" to see a list of supported commands)
+    shell_start();
+
+    // If the credentials haven't been set in NVS, we will wait here for the user
+    // to input them via the shell.
+    if (!nvs_credentials_are_set())
```

Comment:
We should use the new sample credentials module here

Reply:
Right!